WcaOnRails/spec/controllers/results_submission_controller_spec.rb

```diff
@@ -0,0 +1,98 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe ResultsSubmissionController, type: :request do
+  let(:delegate) { FactoryBot.create :delegate }
+  let(:comp) { FactoryBot.create(:competition, delegates: [delegate]) }
+
+  context "not logged in" do
+    it "redirects to sign in" do
+      get submit_results_edit_path(comp.id)
+      expect(response).to redirect_to(new_user_session_path)
+    end
+  end
+
+  context "logged in as a regular user" do
+    sign_in { FactoryBot.create(:user) }
+
+    it "redirects to home page" do
+      get submit_results_edit_path(comp.id)
+      expect(response).to redirect_to(root_url)
+    end
+  end
+
+  context "logged in as a regular delegate" do
+    sign_in { FactoryBot.create(:delegate) }
+
+    it "redirects to home page" do
+      get submit_results_edit_path(comp.id)
+      expect(response).to redirect_to(root_url)
+    end
+  end
+
+  context "logged in as THE delegate" do
+    let(:user) { comp.delegates.first }
+    let(:submission_message) { "Hello, here are the results" }
+    let(:file_contents) { '{ "results": "good" }' }
+    let(:file) do
+      temp_file = Tempfile.new("sometmpfilename.tmp")
+      temp_file.write(file_contents)
+      temp_file.rewind
+      return Rack::Test::UploadedFile.new(temp_file.path, "application/json")
+    end
+
+    before :each do
+      sign_in user
+    end
```

Comment:
The `sign_in` helper does actually call `before :each` under the hood, see [the definition](https://github.com/thewca/worldcubeassociation.org/blob/1dfe8cdd11cacec86ee64441d4bcbea3cdaeba23/WcaOnRails/spec/support/sign_in_macros.rb#L4-L8). So it should be sufficient to just type `sign_in user`.

Reply:
This might not work. I think you don't have access to `lets` unless you're inside of something like an `it` or a `before` =(

Reply:
It looks like you're right:
```
An error occurred while loading ./spec/requests/results_submission_spec.rb.
Failure/Error: sign_in user
  `user` is not available on an example group (e.g. a `describe` or `context` block).
  It is only available from within individual examples (e.g. `it` blocks) or from constructs that run in the scope of an example (e.g. `before`, `let`, etc).
```